Fix `test-webkitpy webkitflaskpy`

gsnedders opened the pull request:
#### 375f0a128550381f809ffad568989797289dda60
<pre>
Fix `test-webkitpy webkitflaskpy`
<a href="https://bugs.webkit.org/show_bug.cgi?id=261108">https://bugs.webkit.org/show_bug.cgi?id=261108</a>

Reviewed by Jonathan Bedard.

webkitpy.test.finder.DirectoryTree.subpath used to consider everything except
for absolute paths as subpaths, and as such everything was always found as a
subtree of the first tree. This is clearly incorrect. Instead, require it
actually start with the name of the package described by the tree.

While we&apos;re at it, make `starting_directory` required everywhere: the only place
where it was previously None was in finder_unittest, which isn&apos;t helpful.

* Tools/Scripts/webkitpy/test/finder.py:
(_DirectoryTree.__init__):
(_DirectoryTree.subpath):
(Finder.add_tree):
* Tools/Scripts/webkitpy/test/finder_unittest.py:
(FinderTest.setUp):
(FinderTest.test_additional_system_paths):
(FinderTest.test_default_names):
(FinderTest.test_paths):
* Tools/Scripts/webkitpy/test/main.py:
(Tester.add_tree):

Canonical link: <a href="https://commits.webkit.org/267685@main">https://commits.webkit.org/267685@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/582fb8b4f85cc9f385c0e0809170cadc48c7fed5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17196 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17522 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18019 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18985 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16096 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/17389 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20796 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/17664 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18291 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17400 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17742 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14936 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19802 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/17335 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14982 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/15627 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22311 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15981 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15794 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20129 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16385 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13901 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15537 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19906 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2126 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16216 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/13436a69642bbcbcac8b975143089be0b53b4a1a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17412

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/582fb8b4f85cc9f385c0e0809170cadc48c7fed5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17412

gsnedders committed:
Fix `test-webkitpy webkitflaskpy`
https://bugs.webkit.org/show_bug.cgi?id=261108

Reviewed by Jonathan Bedard.

webkitpy.test.finder.DirectoryTree.subpath used to consider everything except
for absolute paths as subpaths, and as such everything was always found as a
subtree of the first tree. This is clearly incorrect. Instead, require it
actually start with the name of the package described by the tree.

While we're at it, make `starting_directory` required everywhere: the only place
where it was previously None was in finder_unittest, which isn't helpful.

* Tools/Scripts/webkitpy/test/finder.py:
(_DirectoryTree.__init__):
(_DirectoryTree.subpath):
(Finder.add_tree):
* Tools/Scripts/webkitpy/test/finder_unittest.py:
(FinderTest.setUp):
(FinderTest.test_additional_system_paths):
(FinderTest.test_default_names):
(FinderTest.test_paths):
* Tools/Scripts/webkitpy/test/main.py:
(Tester.add_tree):

Canonical link: https://commits.webkit.org/267685@main

webkit-commit-queue commented:
Committed 267685@main (375f0a128550): <https://commits.webkit.org/267685@main>

Reviewed commits have been landed. Closing PR #17412 and removing active labels.
